### PR TITLE
Send slack notification on failure

### LINKED ISSUE #5664: Notify slack channel on every build failure
**Description**

#5461 changed slack notifications to only notify on real build result changes. However, we would prefer to be notified as well on every build failure.

### PR DESCRIPTION
**Pull request by korthout**
## Description

This PR makes sure to send slack notifications on build every build failure, next to all changes.

## Related issues

closes #5664 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -294,6 +294,18 @@ pipeline {
...
             }
         }
+        failure {
+            script {
+                if (env.BRANCH_NAME != 'develop' || agentDisconnected()) {
+                    return
+                }
+
+                echo "Send slack message"
+                slackSend(
```
Comment (pihme) [inline Jenkinsfile:304]: In case of a change to failure this will now send two notifications. One here and one below.

**Review by pihme — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (korthout) [reply Jenkinsfile:304]: Makes sense 👍

**Comment by korthout** [comment]
@pihme Please have another look

**Review by pihme — APPROVED** [review]

**Comment by korthout** [comment]
@pihme Interesting failures on branch: https://ci.zeebe.camunda.cloud/blue/organizations/jenkins/zeebe-io%2Fzeebe/detail/korthout-5664-notify-on-failure/2/tests

**Pull request merged**